pkg/zk/mul/mul.go

```diff
@@ -50,9 +51,9 @@ func NewProof(hash *hash.Hash, public Public, private Private) *Proof {
 
 	prover := public.Prover
 
-	alpha := sample.UnitModN(N)
-	r := sample.UnitModN(N)
-	s := sample.UnitModN(N)
+	alpha := sample.UnitModN(rand.Reader, N)
```

Comment:
The paper has a type here, I think they meant to say alpha in 2^{L+eps}

Reply:
Maybe we should take care of that in a separate PR?

I'll have a look at this proof myself.

Reply:
Yeah, that would make the most sense, having looked at the paper